Source/ASTableView.mm

```diff
@@ -651,15 +651,12 @@ - (nullable NSIndexPath *)indexPathForNode:(ASCellNode *)cellNode waitingIfNeede
 
 - (NSArray<ASCellNode *> *)visibleNodes
 {
-  NSArray *indexPaths = [self visibleNodeIndexPathsForRangeController:_rangeController];
+  NSArray *indexPaths = [self indexPathsForVisibleRows];
   
   NSMutableArray<ASCellNode *> *visibleNodes = [NSMutableArray array];
   for (NSIndexPath *indexPath in indexPaths) {
     ASCellNode *node = [self nodeForRowAtIndexPath:indexPath];
-    if (node) {
```

Comment:
Do you have confidence that this change is safe?  As one data point, there were some very strange behaviors found that this code worked around; although we shouldn't fear any possible regression, I think protecting against an insert: nil crash is pretty reasonable here.

Maybe add an assertion that node should never be nil, and if it is never reported by a developer who makes more active use of ASTableView (notably @eanagel / Yahoo Messenger), then we're probably safe? :)